### PR TITLE
Set metadata

### PR DESCRIPTION
**Pull request by rickle-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobAsyncClient.java [hunk 1]
```diff
@@ -374,6 +375,61 @@ public void getBlobProperties(String containerName,
...
+        storageBlobServiceClient.setBlobMetadata(containerName, blobName, metadata, callback);
+    }
+
+    /**
+     * Changes a blob's metadata. The specified metadata in this method will replace existing metadata. If old values
+     * must be preserved, they must be downloaded and included in the call to this method.
+     *
+     * @param containerName The container name.
+     * @param blobName      The blob name.
+     * @param timeout           The timeout parameter is expressed in seconds. For more information, see
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobAsyncClient.java:400]: nit. formatting

Comment (gapra-msft) [reply sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobAsyncClient.java:400]: I think you can just do Ctrl + Alt + L to format the whole file

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java [hunk 2]
```diff
@@ -1580,6 +1772,25 @@ private static String readAsString(ResponseBody body) {
                                      @Header("x-ms-encryption-key-sha256") String encryptionKeySha256,
                                      @Header("x-ms-encryption-algorithm") EncryptionAlgorithmType encryptionAlgorithm);
 
+        @PUT("{containerName}/{blob}")
+        Call<ResponseBody> setBlobMetadata(@Path("containerName") String containerName,
+                                              @Path("blob") String blob,
+                                              @Query("timeout") Integer timeout,
+                                              @Header("x-ms-meta") Map<String, String> metadata, // TODO: Maybe use @HeaderMap annotation?
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java:1779]: I think here we may want an extra - so like x-ms-meta-

**Review by gapra-msft — APPROVED** [review]
Looks good, just had a couple comments

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobAsyncClient.java [hunk 3]
```diff
@@ -236,6 +244,140 @@ public void cancel(Context context, long transferId) {
...
     }
 
+    /**
+     * Creates a new container within a storage account. If a container with the same name already exists, the operation
+     * fails.
+     *
+     * @param containerName The container name.
+     * @param callback      Callback that receives the response.
+     */
+    public void createContainer(String containerName,
```
Comment (vcolin7) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobAsyncClient.java:254]: I don't know why the PR includes all a ton of other things, I guess I should just ignore them? I assume this PR only introduces the `setBlobMetadata` operations, correct?

File: sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java [hunk 4]
```diff
@@ -260,6 +475,107 @@ void getBlobProperties(String containerName,
...
+            null,
+            CancellationToken.NONE);
+
+        return blobSetHttpHeadersResponse.getValue();
+    }
+
+    void setBlobMetadata(String containerName,
+                            String blobName,
+                            Map<String, String> metadata,
+                            CallbackWithHeader<Void, BlobSetMetadataHeaders> callback) {
```
Comment (vcolin7) [inline sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java:497]: nit; Indentation.
```suggestion
    void setBlobMetadata(String containerName,
                         String blobName,
                         Map<String, String> metadata,
                         CallbackWithHeader<Void, BlobSetMetadataHeaders> callback) {
```

**Pull request merged**